File: WoodApp-UI/src/app/models/components/pricing-grid.ts
export interface PricingGrid {
    id: number;
    counterpartid: number;
    merchandiseid?: number;
    articleid?: number; // Linked Article ID
    merchandisename?: string;
    merchandisereference?: string;
    discountrate: number;
    validfrom?: Date | string;
    validuntil?: Date | string;
    isactive: boolean;
    notes?: string;
    updatedbyid?: number;
}

export interface PricingGridLookup {
    merchandiseid: number;
    discountrate: number;
}

File: wood-app.api/src/ms.webapp.api.acya.core/Entities/PricingGrid.cs
using ms.webapp.api.acya.core.Interfaces;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ms.webapp.api.acya.core.Entities
{
    public class PricingGrid : IEntity, IAuditable
    {
        public int Id { get; set; }

        public int CounterPartId { get; set; }
        public CounterPart? CounterPart { get; set; }

        public int? MerchandiseId { get; set; }
        public Merchandise? Merchandise { get; set; }

        public double DiscountRate { get; set; }

        public DateTime? ValidFrom { get; set; }
        public DateTime? ValidUntil { get; set; }

        public bool IsActive { get; set; } = true;

        public string? Notes { get; set; }

        public DateTime? CreationDate { get; set; }
        public DateTime? UpdateDate { get; set; }

        public int UpdatedById { get; set; }
        public AppUser? AppUsers { get; set; }

        public PricingGrid()
        {
        }
    }
}
